app/scripts/services/dimStoreService.factory.js

```diff
@@ -476,20 +489,47 @@
           });
         });
 
-        createdItem.hasReforgeNode = !_.isEmpty(reforgeNodes);
-      }
+        // lets just see only the activated nodes for this item instance.
+        var activated = _.filter(item.nodes, function(node) {
+          return node.isActivated;
+        });
 
-      _.each(createdItem.perks, function(perk) {
-        var perkDef = perkDefs.data[perk.perkHash];
-        if (perkDef) {
-          _.each(['displayName', 'displayDescription'], function(attr) {
-            if (perkDef[attr]) {
-              perk[attr] = perkDef[attr];
+        // loop over the exclusive set talents grid for that weapon type
+        _.each(talents.exclusiveSets, function(set) {
+          _.each(activated, function(active) {
+            if(set.nodeIndexes.indexOf(active.nodeHash) > -1) {
+
+              var node = talents.nodes[active.nodeHash].steps[active.stepIndex];
+              if(!_.contains(activePerks, node.nodeStepName)) {
+                createdItem.perks.push({
+                  'displayName': node.nodeStepName,
+                  'displayDescription': node.nodeStepDescription,
+                  'iconPath': node.icon,
+                  'isActive': true,
+                  'order': active.nodeHash
+                });
+              } else {
+                var perk = _.findIndex(createdItem.perks, {displayName: node.nodeStepName});
+
+                createdItem.perks[perk].order = active.nodeHash;
+              }
             }
           });
-        }
+        });
+        // other useful information about the item (this has info about reforge/etc)
+        // _.each(talents.independentNodeIndexes, function(set) {
+
+        createdItem.hasReforgeNode = !_.isEmpty(reforgeNodes);
+      }
+
+      // remove inactive perks, with this we actually lose passive perks (like exotic perks)
+      createdItem.perks = _.filter(createdItem.perks, function(perk) {
+        return perk.isActive;
```

Comment:
Tip for later: I think you could write this as:

```javascript
createdItems.perks = _.filter(createdItem.perks, 'isActive');
```

If you want to keep using the full function predicate, built-in `filter` might be nicer:

```javascript
createdItem.perks = createdItem.perks.filter(function(perk) { return perk.isActive; });
```

Same for `_.each`, etc. Modern JS has a lot of Underscore built-in.